plugin/core/views.py

```diff
@@ -689,3 +690,14 @@ def format_completion(
             details=st_details)
 
     return completion
+
+
+def code_lens_to_phantom(view: sublime.View, code_lens: CodeLens) -> sublime.Phantom:
+    region = range_to_region(Range.from_lsp(code_lens["range"]), view)
+    command = code_lens.get("command")
```

Comment:
the `command` is optional on a code lens :)

```
interface CodeLens {
	/**
	 * The range in which the CodeLens is valid. Should only span a single line.
	 */
	range: Range;

	/**
	 * The command this CodeLens represents.
	 */
	command?: Command;

	/**
	 * A data entry field that is preserved on a CodeLens item between
	 * a CodeLens and a CodeLens resolve request.
	 */
	data?: any
}
```

If no command is provided, 
how will we determine what text to show in the phantom? 
Will we have to expose a method on an AbstractPlugin that will allow a plugin to specify how to convert a code_lensto a phantom?

Reply:
Resolved code lenses **will** have a command. And only resolved code lenses reach this code, AFAIKS.

Reply:
Rafal is right.
```
/**
 * A CodeLense represents a command that should be shown along with
 * source text, like the number of references, a way to run tests, etc.
 *
 * A CodeLens is _unresolved_ when no command is associated to it.
 * For performance reasons, the creation of a CodeLens and resolving should
 * be done in two stages.
 */
```